src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,6 +392,170 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
 };
 
 
+/*
+  This class handles the neural net computation in parallel.
+
+  It can accept just input features, or input features plus iVectors.  */
+class BatchNnetComputer {
+ public:
+  /**
+     This constructor only set up some constant properties.
+     We will use AcceptInput() function to collect the real data.
+     
+     Note: it stores references to all arguments to the constructor, so don't
+     delete them till this goes out of scope.
+
+     @param [in] opts   The options class.  Warning: it includes an acoustic
+                        weight, whose default is 0.1; you may sometimes want to
+                        change this to 1.0.
+     @param [in] nnet   The neural net that we're going to do the computation with
+     @param [in] priors Vector of priors-- if supplied and nonempty, we subtract
+                        the log of these priors from the nnet output.
+     @param [in] compiler  A pointer to the compiler object to use-- this enables the
+                        user to maintain a common object in the calling code that
+                        will cache computations across decodes.  Note: the compiler code
+                        has no locking mechanism (and it would be tricky to design one,
+                        as we'd need to lock the individual computations also),
+                        so the calling code has to make sure that if there are
+                        multiple threads, they do not share the same compiler
+                        object.
+     @param [in] online_ivector_period If you are using iVectors estimated 'online'
+                        (i.e. if online_ivectors != NULL) gives the periodicity
+                        (in frames) with which the iVectors are estimated.
+     @param [in] minibatch_size The capacity of the batch. In general, it means
+                        the number of chunks will be process.
+     @param [in] max_capacity The number of feature matrixes can be stored in
+                        this class. In the case of "very small batch size" or
```

Comment:
I don't feel like this comment really explains what max_capacity does and how the class interprets it.
Actually I don't see why it would be necessary.  And these feature matrices-- do these correspond to utterances, or chunks of utterances?